pandas/core/strings.py

```diff
@@ -464,11 +464,9 @@ def rep(x, r):
         return result
 
 
-def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
+def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=None):
```

Comment:
> no reason to keep a dead API around.

To be clear, this is *no* dead API. Although it is ignored after this PR, everybody using this function uses that keyword. 
So I certainly won't raise (FutureWarning is fine, probably even better as UserWarning anyway)